phoenix-core/src/it/java/org/apache/phoenix/end2end/EmptyColumnIT.java

```diff
@@ -0,0 +1,925 @@
+ *
+ *  These tests are to validate our expectations of the EMPTY_COLUMN behavior.
+ *  Checks :-
+ *  1. Behavior for various tables, views (global and tenant), indexes (global and local)
+ *  2. Behavior under different CF's combinations.
+ *  3. Behavior under different COLUMN ENCODINGS
+ */
+public class EmptyColumnIT extends ParallelStatsDisabledIT {
+
+	private static final Logger logger = LoggerFactory.getLogger(EmptyColumnIT.class);
+	private static int DEFAULT_NUM_ROWS = 5;
+
+	private static class ExpectedTestResults {
+		int numTableRowsExpected;
+		int numTableIndexRowsExpected;
+		int numViewIndexRowsExpected;
+
+		public ExpectedTestResults(int numTableRowsExpected,
+								   int numTableIndexRowsExpected,
+								   int numViewIndexRowsExpected) {
+			this.numTableRowsExpected = numTableRowsExpected;
+			this.numTableIndexRowsExpected = numTableIndexRowsExpected;
+			this.numViewIndexRowsExpected = numViewIndexRowsExpected;
+		}
+	}
+
+    // Scans the HBase rows directly for the empty column and asserts
+	private void assertAllHBaseRowsHaveEmptyColumnCell(byte[] hbaseTableName, byte[] emptyColumnFamilyName, byte[] emptyColumnName, long minTimestamp, int expectedRows) throws IOException, SQLException {
+
+		try (Table tbl = driver.getConnectionQueryServices(getUrl(), TestUtil.TEST_PROPERTIES)
+							 .getTable(hbaseTableName)) {
+
+			Scan allRows = new Scan();
+			allRows.setMaxVersions(1);
+			allRows.setTimeRange(minTimestamp, minTimestamp + Integer.MAX_VALUE);
+			ResultScanner scanner = tbl.getScanner(allRows);
+			int numMatchingRows = 0;
+			for (Result result = scanner.next();  result != null; result = scanner.next()) {
+				numMatchingRows += result.containsColumn(emptyColumnFamilyName, emptyColumnName) ? 1 : 0;
+			}
+
+			assertEquals(String.format(
+					"Expected rows do match for table = %s at timestamp %d",
+					Bytes.toString(hbaseTableName),
+					minTimestamp), expectedRows, numMatchingRows);
+		}
+
+	}
+
+	/**
+	 * -----------------
+	 * Test methods
+	 * -----------------
+	 */
+
+
+
+	@Ignore("fails with java.lang.ArrayIndexOutOfBoundsException: -1 , fails for both Table and GlobalView indexes")
+	// https://issues.apache.org/jira/browse/PHOENIX-5317
+	// https://issues.apache.org/jira/browse/PHOENIX-5322
+	public void testWithBasicTenantViewAndTableIndex() throws Exception {
+
+		// Define the test schema.
+		OtherOptions otherOptions = new OtherOptions();
+		otherOptions.setTableCFs(Lists.newArrayList(null, "A", "B"));
+
+		TableOptions tableOptions = TableOptions.withDefaults();
+		tableOptions.getTablePKColumns().add("ID");
+		tableOptions.getTablePKColumnTypes().add("CHAR(15)");
+
+		GlobalViewOptions globalViewOptions = new GlobalViewOptions();
+		globalViewOptions.setGlobalViewColumns(GLOBAL_VIEW_COLUMNS);
+		globalViewOptions.setGlobalViewColumnTypes(COLUMN_TYPES);
+		otherOptions.setGlobalViewCFs(Lists.newArrayList(null, "A", "B"));
```

Comment:
nit: can we move `otherOptions.setGlobalViewCFs(Lists.newArrayList(null, "A", "B"));` to line 134 for better readabiltiy?